### ISSUE TITLE
[Custom Plugin] When creating  base styles with fontSize property throws error when defaultLineHeights is enabled

### ISSUE THREAD

**Issue by fitimvata** [tailwindlabs/tailwindcss#2308]
Example from docs base styles.
[https://tailwindcss.com/docs/plugins#adding-base-styles](https://tailwindcss.com/docs/plugins#adding-base-styles)

When creating custom plugin  post css throws error.
```
Unknown error from PostCSS plugin. Your current PostCSS version is 7.0.32, but postcss-functions uses 6.0.23. Perhaps this is the source of the error below.

value.charCodeAt is not a function
```

### Config for minimal reproduction:

Version 1.7.6

```
// tailwind.config.js
const plugin = require('tailwindcss/plugin')

module.exports = {
  future: {
    removeDeprecatedGapUtilities: true,
  },
  experimental: {
    applyComplexClasses: true,
    uniformColorPalette: true,
    extendedSpacingScale: true,
    defaultLineHeights: true,
    extendedFontSizeScale: true,
  },
  plugins: [
    plugin(function({ addBase, config }) {
      addBase({
        'h1': { fontSize: config('theme.fontSize.2xl') },
        'h2': { fontSize: config('theme.fontSize.xl') },
        'h3': { fontSize: config('theme.fontSize.lg') },
      })
    })
  ]
}
```

To fix this issue i need to do this:
```
// tailwind.config.js
const plugin = require('tailwindcss/plugin')

module.exports = {
  future: {
    removeDeprecatedGapUtilities: true,
  },
  experimental: {
    applyComplexClasses: true,
    uniformColorPalette: true,
    extendedSpacingScale: true,
    defaultLineHeights: true,
    extendedFontSizeScale: true,
  },
  plugins: [
    plugin(function({ addBase, config }) {
      addBase({
        'h1': { fontSize: config('theme.fontSize.2xl')[0] },
        'h2': { fontSize: config('theme.fontSize.xl')[0] },
        'h3': { fontSize: config('theme.fontSize.lg')[0] },
      })
    })
  ]
}
```
If I disable experimental feature `defaultLineHeights` it works fine.

**Comment by adamwathan** [comment]
I am pretty sure this was fixed, are you on latest version (1.7.6)?

**Comment by fitimvata** [comment]
Yes I'm using the latest version 1.7.6
<img width="755" alt="Screenshot 2020-09-04 at 1 26 03 AM" src="https://user-images.githubusercontent.com/12052390/92183082-9bae3800-ee4d-11ea-91be-62bc444f447c.png">

**Comment by jasonlbeggs** [comment]
I'm running into this issue as well running Tailwind v1.8.13. I'm happy to provide any further details you might need for troubleshooting. Just let me know!

**Comment by shiftgeist** [comment]
Getting the same error with `tailwindcss@1.9.2`, `postcss@7.0.27` and `postcss-functions@3.0.0`. Removing `defaultLineHeights: true` from the `future` configuration objects resolves the issue for me.

**Comment by adamwathan** [comment]
This is fixed in v1.9.5 *but* you have to use the `theme` function, not the `config` function:

```
plugins: [
    plugin(function({ addBase, theme }) {
      addBase({
        'h1': { fontSize: theme('fontSize.2xl') },
        'h2': { fontSize: theme('fontSize.xl') },
        'h3': { fontSize: theme('fontSize.lg') },
      })
    })
  ]
```

Thanks for reporting!

Here's the commit: https://github.com/tailwindlabs/tailwindcss/commit/15ae28a37edbdd6ed848eb7561d0991a29da0a80

**Comment by dargmuesli** [comment]
Can I just swap any `config('theme.` with `theme('` in my plugin's definition?
